src/main/java/com/codeborne/selenide/impl/IO.java

```diff
@@ -0,0 +1,62 @@
+package com.codeborne.selenide.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class IO {
+  private static final Logger log = LoggerFactory.getLogger(IO.class);
+
+  public void copyFile(File sourceFile, File targetFile) throws IOException {
+    try (FileInputStream in = new FileInputStream(sourceFile)) {
+      writeToFile(in, targetFile);
+    }
+  }
+
+  private void writeToFile(InputStream in, File targetFile) throws IOException {
+    ensureFolderExists(targetFile);
+
+    try (FileOutputStream out = new FileOutputStream(targetFile)) {
+      byte[] buffer = new byte[1024];
+      int len;
+      while ((len = in.read(buffer)) != -1) {
+        out.write(buffer, 0, len);
+      }
+    }
+  }
+
+  public void writeToFile(String content, File targetFile) {
+    try (ByteArrayInputStream in = new ByteArrayInputStream(content.getBytes(UTF_8))) {
+      writeToFile(in, targetFile);
+    }
+    catch (IOException e) {
+      log.error("Failed to write file {}", targetFile.getAbsolutePath(), e);
```

Comment:
Also, can you please mind me why don't we throw an exception in such case?

Reply:
@rosolko This code is typically called in case of test failure. We want to save screenshot and page source as an additional information for the error message.

If we throw an exception, then we will loose the original cause of test failure.